### DISCUSSION TITLE
EasyVVUQ Scaling and Performance

### DISCUSSION THREAD

**Discussion by orbitfold** [UCL-CCS/EasyVVUQ#321] [Ideas]
Discuss!

https://ucl-ccs.github.io/EasyVVUQ/dev/bench/

**Comment by orbitfold** [comment]
See https://github.com/UCL-CCS/EasyVVUQ/blob/dev/tests/test_db_benchmark.py for an example of how to do benchmarks with pytest

**Comment by djgroen** [comment]
Hi @orbitfold. I do think this is extremely useful, and this will also be beneficial to benchmark the multi-threaded job submission of FabSim3, which is similarly performance critical. :)
I'd like to include Hamid, but he's not in this repo ;).

**Reply by orbitfold** [reply]
ok I'll add him

**Reply by djgroen** [reply]
I notified Hamid of this activity now. But before we can use this for FabSim3 we are actually going to port the multithreading implementation to multiprocessing.

**Reply by orbitfold** [reply]
I recommend you look into python futures https://rednafi.github.io/digressions/python/2020/04/21/python-concurrent-futures.html

**Comment by DavidPCoster** [comment]
![image](https://user-images.githubusercontent.com/40796322/109983190-e3315600-7d02-11eb-9f46-0b95e95389dd.png)
Plot of time taken for a PCE campaign with increasing PCE order.  
I have plotted the total time and then the time taken for the three most expensive phases.

**Reply by DavidPCoster** [reply]
Results collation seems to be scaling with the square of the number of points, and the analysis phase even faster ...

**Comment by orbitfold** [comment]
https://www.whoishostingthis.com/compare/sqlite/optimize/

Some SQLite optimisation ideas. Some of them we already do. But the caching stuff is interesting to look into.
